lightning/src/ln/channelmanager.rs

```diff
@@ -5173,49 +5190,54 @@ where
 		let mut peer_state_lock = peer_state_mutex.lock().unwrap();
 		let peer_state = &mut *peer_state_lock;
 		let is_only_peer_channel = peer_state.total_channel_count() == 1;
-		match peer_state.inbound_v1_channel_by_id.entry(temporary_channel_id.clone()) {
-			hash_map::Entry::Occupied(mut channel) => {
-				if !channel.get().is_awaiting_accept() {
-					return Err(APIError::APIMisuseError { err: "The channel isn't currently awaiting to be accepted.".to_owned() });
+
+		// Find (and remove) the channel in the unaccepted table. If it's
+		// not there, something weird is happening and return an error.
+		let mut channel = match peer_state.inbound_channel_request_by_id.remove(temporary_channel_id) {
+			Some(unaccepted_channel) => {
+				let best_block_height = self.best_block.read().unwrap().height();
+				InboundV1Channel::new(&self.fee_estimator, &self.entropy_source, &self.signer_provider,
+					counterparty_node_id.clone(), &self.channel_type_features(), &peer_state.latest_features,
+          &unaccepted_channel.open_channel_msg, user_channel_id, &self.default_configuration, best_block_height,
+					&self.logger, unaccepted_channel.outbound_scid_alias).map_err(|e| APIError::ChannelUnavailable { err: e.to_string() })
+			}
+			_ => Err(APIError::APIMisuseError { err: "The channel isn't currently awaiting to be accepted.".to_owned() })
```

Comment:
I think this error message could be a little confusing now, because at this stage we don't even know if we have that channel at all (it could be in the other sets but maybe not). So it could also indicate non-existence.

Reply:
Gotcha... to be fair the user _is_ calling `accept_inbound_channel(_from_trusted_peer_0conf)?` so the _user_ probably thinks that there is a channel waiting to be accepted. What might be clearer here? Maybe `No such channel awaiting to be accepted`?

Reply:
Yeah, I think that sounds more accurate IMO :)